docs/multicluster/architecture.md

```diff
@@ -0,0 +1,94 @@
+across multiple Kubernetes clusters.
+
+The below diagram depicts a basic multi-cluster topology in Antrea.
+
+<img src="assets/basic-topology.png" width="400" alt="Antrea Multi-cluster Topology">
+
+Given a set of Kubernetes clusters, there will be a leader cluster and several member clusters.
+By default, a leader cluster itself is also a member cluster of a ClusterSet. A cluster
+can also be configured as a dedicated leader cluster of multiple ClusterSets.
+
+## Terminology
+
+ClusterSet is a placeholder name for a group of clusters with a high degree of mutual
+trust and shared ownership that share Services amongst themselves. Within a ClusterSet,
+Namespace sameness applies, which means all Namespaces with a given name are considered to
+be the same Namespace. The ClusterSet Custom Resource Definition(CRD) defines a ClusterSet
+including the leader and member clusters information.
+
+The ClusterClaim CRD is used to claim a cluster itself as a member of a ClusterSet with
+unique cluster ID, to claim a ClusterSet with unique ClusterSet ID.
+
+The MemberClusterAnnounce CRD declares a member cluster configuration to the leader cluster.
+
+The Common Area is an abstraction in the Antrea Multi-cluster implementation provides a storage
+interface for resource export/import that can be read/written by all member and leader clusters
+in the ClusterSet. The Common Area is implemented with a Namespace in the leader cluster
+
+## Antrea Multi-cluster Controller
+
+In a member cluster, Antrea Multi-cluster creates a Deployment that runs Antrea Multi-cluster
+Controller which is responsible for exporting resource to and importing resource from a leader
+cluster in a ClusterSet.
+
+In a leader cluster, Antrea Multi-cluster creates a Deployment that runs Antrea Multi-cluster
+Controller which is responsible for converting resources from different member clusters into one
+encapsulated resource as long as these resources have the same kind and match Namespace sameness.
+
+In ClusterSet initialization, Antrea Multi-cluster Controller in a member cluster watches
+ClusterSet, ClusterClaim, and creates a MemberClusterAnnounce in the leader cluster.
+
+For resource exporting, it watches ServiceExports, Services and Endpoints resources, encapsulates
+Services and Endpoints into ResourceExports according to ServiceExports, and writes the ResourceExports
+to leader cluster. For resource importing, it watches ResourceImports from leader cluster, and
+creates multi-cluster Services and Endpoints with a prefix `antrea-mc-` plus exported Service name,
+and also ServiceImports which have the same name as the Service name.
+
+In a leader cluster, for ClusterSet initialization, Antrea Multi-cluster controller watches
+ClusterSet, ClusterClaim, and validate the ClusterSet and Clusterclaim. For resource export/import,
```

Comment:
watches and validates ClusterSet and ClusterClaims

Reply:
done